From eb3eb90be8b81bb695c0c2b9932bc70fbd10376c Mon Sep 17 00:00:00 2001
From: Tony Camuso <tcamuso@redhat.com>
Date: Fri, 8 Aug 2014 18:14:46 -0400
Subject: [char] ipmi: Turn off default probing of interfaces

Message-id: <1407521687-4858-13-git-send-email-tcamuso@redhat.com>
Patchwork-id: 87555
O-Subject: [RHEL7.1 PATCH 12/13] ipmi: Turn off default probing of interfaces
Bugzilla: 1111214
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1111214

Cherry-picked without conflicts from the following upstream commit.

commit 0dfe6e7ed47feeb22f3cf8c7d8ac7e65bd4e87f5
Author: Corey Minyard <cminyard@mvista.com>
Date:   Mon Apr 14 09:46:53 2014 -0500

    ipmi: Turn off default probing of interfaces

    The default probing can cause problems with some system, slow booting,
    extra CPU usages, etc.  Turn it off by default and give a config option
    to enable it.

    From: Matthew Garrett <matthew.garrett@nebula.com>
    Signed-off-by: Corey Minyard <cminyard@mvista.com>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

Signed-off-by: Tony Camuso <tcamuso@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/char/ipmi/Kconfig b/drivers/char/ipmi/Kconfig
index 0baa8fa..db1c9b7 100644
--- a/drivers/char/ipmi/Kconfig
+++ b/drivers/char/ipmi/Kconfig
@@ -50,6 +50,18 @@ config IPMI_SI
   Currently, only KCS and SMIC are supported.  If
   you are using IPMI, you should probably say "y" here.
 
+config IPMI_SI_PROBE_DEFAULTS
+       bool 'Probe for all possible IPMI system interfaces by default'
+       default n
+       depends on IPMI_SI
+       help
+  Modern systems will usually expose IPMI interfaces via a discoverable
+  firmware mechanism such as ACPI or DMI. Older systems do not, and so
+  the driver is forced to probe hardware manually. This may cause boot
+  delays. Say "n" here to disable this manual probing. IPMI will then
+  only be available on older systems if the "ipmi_si_intf.trydefaults=1"
+  boot argument is passed.
+
 config IPMI_WATCHDOG
        tristate 'IPMI Watchdog Timer'
        help
diff --git a/drivers/char/ipmi/ipmi_si_intf.c b/drivers/char/ipmi/ipmi_si_intf.c
index 7af21dc..e712966 100644
--- a/drivers/char/ipmi/ipmi_si_intf.c
+++ b/drivers/char/ipmi/ipmi_si_intf.c
@@ -1231,7 +1231,7 @@ static bool          si_tryplatform = 1;
 #ifdef CONFIG_PCI
 static bool          si_trypci = 1;
 #endif
-static bool          si_trydefaults = 1;
+static bool          si_trydefaults = IS_ENABLED(CONFIG_IPMI_SI_PROBE_DEFAULTS);
 static char          *si_type[SI_MAX_PARMS];
 #define MAX_SI_TYPE_STR 30
 static char          si_type_str[MAX_SI_TYPE_STR];
-- 
1.7.1